maxim/logger/parsers/generation_parser.py

```
@@ -219,8 +220,20 @@ def default_json_serializer(o: Any) -> Any:
     """
     if isinstance(o, enum.Enum):
         return o.value
+    # Read-only dict wrappers (e.g. a class' __dict__) are not JSON serializable
+    # by default. This commonly shows up when response_format is passed as a
+    # Pydantic model / class instead of a plain dict.
+    if isinstance(o, types.MappingProxyType):
+        return dict(o)
     if hasattr(o, "to_dict"):
         return o.to_dict()
+    # Pydantic v2 models expose model_dump(); v1 models expose dict().
+    if hasattr(o, "model_dump") and callable(getattr(o, "model_dump")):
```

Ruff (0.15.21)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)